lib/ts/station_fsm.ex

```diff
@@ -223,16 +198,18 @@ defmodule StationFsm do
 	# start state
 	defstate start do
 		# On getting the data input, go to ready state
-		defevent input_data(station_vars, registry, qc) do
-			vars = [station_vars, registry, qc]
-			next_state(:ready, vars)
+		defevent input_data(station_data = [_station_vars,
+		{_registry, _qc, _station}])
+		do
+			next_state(:ready, station_data)
 		end
 	end
 
 	# ready state
 	defstate ready do
 		# When local variables of the station are updated
-		defevent update(new_vars), data: original_vars do
+		defevent update(new_vars), data: [_station_vars,
```

Comment:
@SuhasHebbar you might want to articulate a consistent set of rules for names and put them on  https://github.com/prasadtalasila/TransportScheduler/wiki/CS-Names.